### ISSUE TITLE
Renaming path of a git-repo fails to deploy changes due to helm owner changes

### ISSUE THREAD

**Issue by Martin-Weiss** [rancher/fleet#502]
We have a directory with YAML files in git that is applied via fleet.
For this we have defined a git repo with a subdirectory and without a fleet.yaml.

Now we had to change the path and the  update done by fleet fails due to a change in the helm deployment done by fleet for the yaml files..

Problem seems to be that the bundle / helm release name is build based on the git-repo and path name..

Workaround can be to add the helm: releaseName: .. in fleet.yaml but in general basing the helm release name on a path might not be the best choice and instead we should have a persistent name randomly created in case the releaseName is not defined in fleet.yaml.

Oh - and currently it seems the "fleet always does helm" and "fleet should have fleet.yaml with helm: releaseName:" specified even with pure non helm chart based deployments is not documented properly or I might have overseen it ;-).

**Comment by Martin-Weiss** [comment]
FYI - we have added a releaseName but for some reason the repo is now in "out of sync" status. It seems that also the bundle names are related to the paths in the git repo and changing paths results in some problem with the bundle names.

Any idea what we can do to get this cleaned up?

![grafik](https://user-images.githubusercontent.com/13362573/133212400-20a6f5af-a509-4afd-aaee-aea9148d1f98.png)

**Comment by StrongMonkey** [comment]
@Martin-Weiss Are you able to fix this by specifying releaseName in the end? Chaning the path of git-repo is not supported currently. Out of sync means that some of the cluster is not able to deploy/sync the application.

**Comment by Martin-Weiss** [comment]
> @Martin-Weiss Are you able to fix this by specifying releaseName in the end?

The current "workaround" is to
1. specify the releaseName in fleet.yaml
2. create a fleet.yaml in each empty directory
3. not have any files in the git repo that are not required

In addition to this we basically had to uninstall everything with "helm delete" and then "re-install" - which is a big challenge for CRDs and Longhorn.

>Changing the path of git-repo is not supported currently. 

This might be a usability, a UI and a documentation issue.
As we have learned - this happens - and the result is critical - so I believe we have to enhance the UI, add documentation and fleet should detect and block such a change if it is not supported.

>Out of sync means that some of the cluster is not able to deploy/sync the application.

Just wild guessing - I believe we ran into this issue because we did changes before fleet could complete them.
Not sure if this is possible to be reproduced but it might be related to path change and do an other change / git push before the first one got completed. Not sure how fleet works with multiple changes going on regarding keeping the sequence..
